### ISSUE TITLE
ParamConverterProvider not being picked up

### ISSUE THREAD

**Issue by alesj** [quarkusio/quarkus#22287]
### Describe the bug

I have a ParamConverterProvider on the REST client interface registered 

```java
@RegisterRestClient(configKey = "qcltest")
@RegisterProvider(RequestFilter.class)
@RegisterProvider(CTypeConverterProvider.class)
public interface TestService {
    @GET
    @Path("/{type}")
    Fact fact(
        @PathParam("type") CType type
    );
}
```
but it's not being used

### Expected behavior

CTypeConverterProvider should be used.

### Actual behavior

CType is written as Ctype::toString aka Object::toString

### How to Reproduce?

https://github.com/alesj/quarkus_cl130/tree/convert1

Simply run Main class or `mvn quarkus:dev`

### Quarkus version or git rev

2.5.2.Final

**Comment by alesj** [comment]
@michalszynkiewicz @geoand ^

**Comment by geoand** [comment]
@michalszynkiewicz looking at https://github.com/quarkusio/quarkus/blob/2.6.0.Final/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/jaxrs/ConfigurationImpl.java#L219 I see that `ParamConverterProvider` is not handled at all. I assume the TCK doesn't test for it so support for it was not added?

**Comment by michalszynkiewicz** [comment]
I think so

**Comment by alesj** [comment]
Hmmm, I know this worked before ... when I had that classic rest-client.
When moving to reactive, it stopped.

**Comment by gsmet** [comment]
@michalszynkiewicz I think we should try to fix this for 2.7 as we really want to make a big push on using the RESTEasy Reactive stack.

**Comment by michalszynkiewicz** [comment]
i already started working on it
